src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -628,5 +628,27 @@ private static void StelemRef_Helper_NoCacheLookup(ref object? element, void* el
 
             throw new ArrayTypeMismatchException();
         }
+
+        [DebuggerHidden]
+        [StackTraceHidden]
+        [DebuggerStepThrough]
+        private static bool SimpleIsInstanceOf(void* typeHnd, object? obj)
```

Comment:
This helper is not perf critical. It would be better to keep it in C++ and just make it call `ExceptionIsOfRightType`. It would ensure that the generic and non-generic paths stay in sync.